lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +26,151 @@ def fragment_cache(cached_hash, non_cached_hash)
           Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
         end
 
+        private
+
+        def serialize_association(association)
+          serialized_or_virtual_of(association.serializer, association.options)
+        end
+
+        # iterate through the associations on the serializer,
+        # adding them to the parent as needed (as singular or plural)
+        #
+        # restrict_to is a list of symbols that governs what
+        # associations on the passed in seralizer to include
+        def add_resource_relationships(parent, serializer, restrict_to = [])
```

Comment:
`restrict_to` could be renamed to something else that could express it better.
We are facing it as restricting because our mindset on #1113 , but you are actually listing what you want, this is a non-blocking comment, just some details that might help when new ppl jump into this :smile: